### ISSUE TITLE
import-blacklist rule false positive for dynamic imports

### ISSUE THREAD

**Issue by use-strict** [palantir/tslint#4307]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.0.1
- __Running TSLint via__: tslint-loader, vscode-typescript-tslint-plugin

#### TypeScript code being linted

```ts
let format: import("bignumber.js").BigNumber.Format = {
    groupSize: 3
};
```

with `tslint.json` configuration:

```json
{
    "extends": "tslint:recommended",
    "rules": {
        "align": [true,
            "statements"
        ],
        "arrow-return-shorthand": true,
        "arrow-parens": false,
        "array-type": [
            true,
            "array"
        ],
        "callable-types": false,
        "comment-format": [false,
            "check-space"
        ],
        "import-spacing": true,
        "interface-name": true,
        "member-access": [
            false
        ],
        "member-ordering": [true, {
            "order": [
                "static-field",
                "instance-field",
                "static-method",
                "constructor",
                "instance-method"
            ]
        }],
        "no-angle-bracket-type-assertion": true,
        "no-console": [true, "log", "info", "warn", "error"],
        "no-empty-interface": false,
        "no-default-export": true,
        "no-inferrable-types": [true],
        "no-string-throw": true,
        "object-literal-sort-keys": false,
        "ordered-imports": [false,
            {
                "import-sources-order": "case-insensitive",
                "named-imports-order": "case-insensitive"
            }
        ],
        "prefer-const": false,
        "prefer-method-signature": true,
        "quotemark": [true, "double"],
        "trailing-comma": [
            true,
            {
                "multiline": "never",
                "singleline": "never"
            }
        ],
        "variable-name": [true, "ban-keywords", "check-format", "allow-leading-underscore", "allow-pascal-case"],
        "import-blacklist": [true, "bignumber.js", "@sentry/browser"],
        "whitespace": [true,
            "check-branch",
            "check-decl",
            "check-operator",
            "check-separator",
            "check-rest-spread",
            "check-type",
            "check-type-operator",
            "check-preblock"
        ]
    }
}

```

#### Actual behavior
TSLINT: This import is blacklisted, import a submodule instead (import-blacklist)

#### Expected behavior
There should be no errors for dynamically importing types. The rule used to work as expected in tslint 4, but it seems it was refactored multiple types (the main suspect is https://github.com/palantir/tslint/pull/2613). This prevents me from upgrading to tslint 5.

Clarification: The purpose of this rule is to prevent accidental inclusion of a big library in the app bundle. Dynamic imports for types only have an effect at compile time and shouldn't result in an error. It can also be argued that dynamic imports of any kind should not result in errors as they are used in conjunction with code splitting.

**Comment by JoshuaKGoldberg** [comment]
> This prevents me from upgrading to tslint 5

Thanks for filing this @use-strict! That sounds serious. Can you disable the rule with an `"import-blacklist": false` line in your `tslint.json` `"rules"`?

> There should be no errors for dynamically importing types

Shouldn't there be? What if someone does this in your code?

```typescript
import("evil.js").then(() => { /* ... */ });
```

Should the rule be able to indicate modules that can only be lazy-loaded?

**Comment by adidahiya** [comment]
> The purpose of this rule is to prevent accidental inclusion of a big library in the app bundle.

I don't think this is the only purpose of this rule. As stated in the documentation, it is more generic than that:

> In other cases, you may simply want to ban an import because using it is undesirable or unsafe.

It's unfortunate that this ended up being a regression for your usage of TSLint, but I'm not inclined to relax the behavior by default. Perhaps someone could add a new rule option to ignore type-only imports of blacklisted modules.

**Comment by use-strict** [comment]
Yes, it is unfortunate, especially since I was the one who proposed the rule and its purpose was initially only that which I stated (see https://github.com/palantir/tslint/pull/1549).

> In other cases, you may simply want to ban an import because using it is undesirable or unsafe.
Although this looks to me like an over-generalization with little real world support, I must agree with you this description implies no import, even type-only, should be allowed.

The logical conclusion would be to indeed add a rule option to relax the behavior. I'll see if I have some time to make another PR. If you have a naming suggestion for the option, it would be great :)

**Comment by JoshuaKGoldberg** [comment]
`allow-import-types` as a first attempt at a name? Not too attached to that...

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!
